src/main/java/clients/symphony/api/DatafeedClientV1.java

```diff
@@ -0,0 +1,127 @@
+            } else {
+                StringId datafeedId = response.readEntity(StringId.class);
+                logger.info("Created new datafeed {} for bot {}", datafeedId.getId(),
+                    botClient.getBotUsername());
+
+                // if the reuseDatafeedID config isn't set (null), we assume its default value as true
+                if (botClient.getConfig().getReuseDatafeedID() == null || botClient.getConfig().getReuseDatafeedID()) {
+                    writeDatafeedIdToDisk(botClient.getConfig(), datafeedId.getId());
+                }
+
+                return datafeedId.getId();
+            }
+        }
+    }
+
+    private void writeDatafeedIdToDisk(SymConfig config, String datafeedId) {
+        String agentHostPort = config.getAgentHost() + ":" + config.getAgentPort();
+
+        try (FileWriter fw = new FileWriter(botClient.getDatafeedIdFile())) {
+            fw.write(datafeedId + "@" + agentHostPort);
+            fw.flush();
+        } catch (IOException ex) {
+            logger.error(ex.getMessage());
+        }
+    }
+
+    @Override
+    public List<DatafeedEvent> readDatafeed(String id) throws SymClientException {
+        WebTarget webTarget = botClient.getAgentClient().target(config.getAgentUrl());
+        Invocation.Builder builder = webTarget
+            .path(AgentConstants.READDATAFEED.replace("{id}", id))
+            .request(MediaType.APPLICATION_JSON)
+            .header("sessionToken", botClient.getSymAuth().getSessionToken())
+            .header("keyManagerToken", botClient.getSymAuth().getKmToken());
+
+        List<DatafeedEvent> datafeedEvents = null;
+
+        logger.debug("Reading datafeed {}", id);
+        try (Response response = builder.get()) {
+            if (response.getStatusInfo().getFamily() != SUCCESSFUL) {
+                logger.error("Datafeed read error for request: {}", webTarget.getUri());
+                handleError(response, botClient);
+            } else if (response.getStatusInfo().getFamily() == CLIENT_ERROR) {
+                ((SymLoadBalancedConfig) config).rotateAgent();
+            } else {
+                if (response.getStatus() == NO_CONTENT.getStatusCode()) {
+                    datafeedEvents = new ArrayList<>();
+                } else {
+                    datafeedEvents = response.readEntity(DatafeedEventsList.class);
+                }
+            }
+            return datafeedEvents;
+        }
+    }
+
+
+    @Override
+    public List<StringId> listDatafeedId() {
+        throw new UnsupportedOperationException("List Datafeed not supported in Datafeed v1");
+    }
+
+    @Override
+    public void deleteDatafeed(String id) {
+        throw new UnsupportedOperationException("Delete Datafeed not supported in Datafeed v1");
+    }
+
+    @Override
+    public List<DatafeedEvent> readDatafeed(String id, String ackId) {
+        throw new UnsupportedOperationException("Read Datafeed with ackId not supported in Datafeed v1");
+    }
```

Comment:
I would actually keep a single `readDatafeed` method but using varargs to define the ackId like : 
```java
public List<DatafeedEvent> readDatafeed(String id, String... ackId) { /*  */ }
```
With this strategy you make your API simpler to implement, to document and to use by developers.